operator/pkg/client/users/client_test.go

```diff
@@ -228,3 +230,98 @@ func TestClientPasswordCreation(t *testing.T) {
 		})
 	}
 }
+
+func TestPasswordNotGenerated(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*2)
+	defer cancel()
+
+	testEnv := testutils.RedpandaTestEnv{}
+	cfg, err := testEnv.StartRedpandaTestEnv(false)
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	err = redpandav1alpha2.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+
+	c, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	require.NoError(t, err)
+	require.NotNil(t, c)
+
+	container, err := redpanda.Run(ctx, redpandaTestContainerImage,
+		redpanda.WithEnableKafkaAuthorization(),
+		redpanda.WithEnableSASL(),
+		redpanda.WithSuperusers("user"),
+		redpanda.WithNewServiceAccount("user", "password"),
+	)
+
+	require.NoError(t, err)
+
+	broker, err := container.KafkaSeedBroker(ctx)
+	require.NoError(t, err)
+
+	admin, err := container.AdminAPIAddress(ctx)
+	require.NoError(t, err)
+
+	kafkaClient, err := kgo.NewClient(kgo.SeedBrokers(broker), kgo.SASL(scram.Auth{
+		User: "user",
+		Pass: "password",
+	}.AsSha256Mechanism()))
+	require.NoError(t, err)
+
+	rpadminClient, err := rpadmin.NewAdminAPI([]string{admin}, &rpadmin.BasicAuth{
+		Username: "user",
+		Password: "password",
+	}, nil)
+	require.NoError(t, err)
+
+	usersClient, err := NewClient(ctx, c, kadm.NewClient(kafkaClient), rpadminClient)
+	require.NoError(t, err)
+	defer usersClient.Close()
+
+	annotations := map[string]string{
+		"test": "annotation",
+	}
+	labels := map[string]string{
+		"test": "label",
+	}
+
+	user := &redpandav1alpha2.User{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testuser" + strconv.Itoa(int(time.Now().UnixNano())),
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: redpandav1alpha2.UserSpec{
+			ClusterSource: &redpandav1alpha2.ClusterSource{
+				ClusterRef: &redpandav1alpha2.ClusterRef{
+					Name: "bogus",
+				},
+			},
+			Authentication: &redpandav1alpha2.UserAuthenticationSpec{
+				Type: ptr.To(redpandav1alpha2.SASLMechanismScramSHA512),
+				Password: redpandav1alpha2.Password{
+					Value: "",
+					ValueFrom: &redpandav1alpha2.PasswordSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "testsecret",
```

Comment:
Can you add a case where the secret exists but `Key` doesn't exist?

Reply:
Thanks added.